src/components/connect.js

```diff
@@ -19,6 +19,14 @@ function getDisplayName(WrappedComponent) {
   return WrappedComponent.displayName || WrappedComponent.name || 'Component'
 }
 
+function getMapDispatch(value) {
+  switch (true) {
+    case (typeof value === 'function'): return value
```

Comment:
The main things I would promote are the function extraction, it adds some more clarity imo, and the revision to using `isPlainObject` as the second guard statement or `else-if`.